client/src/components/comparison/Comparison.jsx

```diff
@@ -1,9 +1,100 @@
-import React from 'react';
+import React, {useEffect, useState/*, useContext*/} from 'react';
+import RelatedProducts from './RelatedProducts.jsx';
+import UserOutfitItems from './UserOutfitItems.jsx';
+// import {products} from '../../../mock-data/products.js';
+// import {productStyles} from '../../../mock-data/productStyles.js';
+import Requester from '../../Requester.js';
+import LoadingSpinner from '../../components/shared/LoadingSpinner.jsx';
 
-const Comparison = (props) => {
-  return (
-    <div>Comparison</div>
-  );
-};
+const rerequester = Requester();
+
+class Comparison extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      relatedProducts: this.props.relatedProducts,
+      detaledRelatedProductsArr: [],
+      relatedProductsStylesArr: [],
+      outfitList: [],
+      reviewsMetaArr: [],
+      activeIndex: 0,
+      styles: this.props.styles
+    };
+  }
+
+  // function that will handle product click
+
+  // function that will handle the compare Modal
+
+  // function that will handle adding an item to the outfit list
+
+  // handleAddOutfitItem = (product, style) => {
+  //   if (!outfitList.some(obj => obj.id === product.id)) {
+  //     this.setState({
+  //       outfitList: [...outfitList, product]});
+  //   }
+  // };
+
+  // function that will handle removing an item form the outfit list
+
+  // get and store the product obj from each product in the relatedProducts array
+  getTheDeets = ()=> {
+    return Promise.all(this.state.relatedProducts.map(relatedProduct => rerequester.getProduct(relatedProduct)))
+      .then((data) => {
+        // console.log('this is working', data);
+        this.setState({
```

Comment:
I'm not sure here, but I believe that because you're not returning anything from this `then` block, the following `then` block may run before you want it to (it may not matter as long as `getTheStyles()` doesn't depend on the result of this `setState` call).

Reply:
It was working before but I still made changes to see how it would effect the function.  It seems to be working significantly faster now.